api/soap/mc_filter_api.php

```diff
@@ -147,3 +191,165 @@ function mc_filter_get_issue_headers( $p_username, $p_password, $p_project_id, $
 
 	return $t_result;
 }
+
+/**
+ * Get all issue headers matching the custom filter.
+ *
+ * @param string                $p_username         The name of the user trying to access the filters.
+ * @param string                $p_password         The password of the user.
+ * @param FilterSearchData      $p_filter_search    The custom filter.
+ * @return array that represents an IssueHeaderDataArray structure
+ */
+function mc_filter_search_issue_headers( $p_username, $p_password, $p_filter_search ) {
+
+    global $_SOAP_API_TO_FILTER_API_NAMES;
+
+    $t_user_id = mci_check_login( $p_username, $p_password );
+
+    if( $t_user_id === false ) {
+        return mci_soap_fault_login_failed();
+    }
+
+    // object to array
+    if( is_object($p_filter_search) ) {
+        $p_filter_search = get_object_vars($p_filter_search);
+    }
+
+    $t_project_id = $p_filter_search['project_id'];
+
+    if( !mci_has_readonly_access( $t_user_id, $t_project_id  ) ) {
+        return mci_soap_fault_access_denied( $t_user_id );
+    }
+
+    $t_filter = array( '_view_type' => 'advanced' );
+
+    // default fields
+    foreach( $_SOAP_API_TO_FILTER_API_NAMES as $t_soap_name => $t_filter_name ) {
+        if ( isset ( $p_filter_search[$t_soap_name]) ) {
+
+            $t_value = $p_filter_search[$t_soap_name];
+            error_log('Simple extraction, value is ' . print_r($t_value, true) . ' .' );
```

Comment:
I would drop these error_logs in favor of log_event() calls (see core/logging_api.php).  This directs the log to MantisBT log files and have the log entries visibility controlled by $g_logging_level (see config_defaults_inc.php) having LOG_FILTERING set.